infra/examples-dev/gcp/msft-365.tf

```diff
@@ -0,0 +1,59 @@
+# BEGIN MSFT
+
+module "worklytics_connectors_msft_365" {
+  source = "../../modules/worklytics-connectors-msft-365"
+  # source = "git::https://github.com/worklytics/psoxy//infra/modules/worklytics-connectors-msft-365?ref=rc-v0.4.36"
+
+
+  enabled_connectors     = var.enabled_connectors
+  environment_id         = var.environment_name
+  msft_tenant_id         = var.msft_tenant_id
+  example_msft_user_guid = var.example_msft_user_guid
+  msft_owners_email      = var.msft_owners_email
+  todo_step              = 1
+}
+
+provider "azuread" {
+  tenant_id = var.msft_tenant_id
+}
+
+locals {
+  env_qualifier           = coalesce(var.environment_name, "psoxy")
+  msft_365_enabled        = length(module.worklytics_connectors_msft_365.enabled_api_connectors) > 0
+  developer_provider_name = "${local.env_qualifier}-azure-access"
+}
+
+module "msft-connection-auth-federation" {
+  for_each = module.worklytics_connectors_msft_365.enabled_api_connectors
+
+  source = "../../modules/azuread-federated-credentials"
+  # source = "git::https://github.com/worklytics/psoxy//infra/modules/azuread-local-cert?ref=v0.4.36"
+
+  application_object_id = each.value.connector.id
+  display_name          = "GcpFederation"
+  description           = "Federation to be used for psoxy Connector - ${each.value.display_name}${local.env_qualifier}"
+  issuer                = "https://accounts.google.com"
+  subject               = module.psoxy.service_accounts[each.key].unique_id
+}
+
+locals {
+  msft_api_connectors_with_auth = {
+    for k, msft_connector in module.worklytics_connectors_msft_365.enabled_api_connectors :
+    k => merge(msft_connector, {
+      environment_variables = merge(try(msft_connector.environment_variables, {}),
+        {
+          # NOTE: hardcoded due a cycle (through api_connectors -> module.psoxy), ideally better if coming from
+          # module.msft-connection-auth-federation[k].audience output variable
+          # but for GCP is always "api://AzureADTokenExchange".
+          AUDIENCE          = "api://AzureADTokenExchange"
```

Comment:
Cleaner to get it from the output, but a big change needs to be done or the approach should be with a different structure than in AWS case. I've done this to keep same approach but having this hardcoded; in theory for GCP it should not change